apps/src/blockly/addons/functionEditor.js

```diff
@@ -215,20 +215,18 @@ export default class FunctionEditor {
         fields: {
           NAME: newProcedure.getName(),
         },
-        deletable: false,
       };
 
       this.block = Blockly.serialization.blocks.append(
         this.addEditorWorkspaceBlockConfig(newDefinitionBlock),
         this.editorWorkspace
       );
     }
+    this.block.setDeletable(false);
 
-    const isBehavior = type === BLOCK_TYPES.behaviorDefinition;
-    // We do not want to show the delete button for behaviors that are not user-created
-    // or are in read only workspaces.
-    const hideDeleteButton =
-      this.isReadOnly || (isBehavior && !this.block.userCreated);
+    // We only want to be able to delete things that are user-created (functions and behaviors)
+    // and not things that are being previewed from a read-only workspace.
```

Comment:
This comment was modified in the cherry-pick since @molly-moen also updated it, and accounts for the extra diff.